.github/workflows/test-build-cygwin.yml

```diff
@@ -0,0 +1,74 @@
+name: Test on Cygwin
+on: ["push", "pull_request"]
```

Comment:
@DWesl
Although we like the idea of CI testing all features we aren't ready for this style. I suggest changing this to
`on: ["workflow_dispatch", "pull_request", "release"]`

Reply:
Done.